Refactor inference benchmark and add runner script to do sweep

mikaylagawarecki opened the pull request:
- Added `runner.sh` that does a sweep over `batch_size=(1, 32, 64, 128, 256)` and `compile=(True, False)`
- Added GPU utilization as a metric
- Converted frontend from 2 processes (one putting requests into `request_queue` and one reading from `response_queue` and collecting metrics) to a single process with 3 threads (one putting requests into `request_queue` and one reading from `response_queue` and collecting metrics and one polling `nvidia-smi` for gpu utilization)

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #112900
* __->__ #112863
* #112582

mikaylagawarecki committed:
Refactor inference benchmark add runner script to do sweep

[ghstack-poisoned]

albanD commented on benchmarks/inference/results/output.txt:
```diff
@@ -0,0 +1,102 @@
+=============================================================================
```
Do you actually want to checkin this file?

mikaylagawarecki replied:
I'll remove it and add the results back when I make the runner output things in tabular form(?)

albanD reviewed (approved):
SGTM!

mikaylagawarecki committed:
Update on "Refactor inference benchmark and add runner script to do sweep"

- Added `runner.sh` that does a sweep over `batch_size=(1, 32, 64, 128, 256)` and `compile=(True, False)`
            - Added results from sweep in `outputs/`
- Added GPU utilization as a metric
- Converted frontend from 2 processes (one putting requests into `request_queue` and one reading from `response_queue` and collecting metrics) to a single process with 3 threads (one putting requests into `request_queue` and one reading from `response_queue` and collecting metrics and one polling nvidia-smi for gpu utilization)




[ghstack-poisoned]

mikaylagawarecki commented:
@pytorchbot merge

mikaylagawarecki commented:
@pytorchbot merge -f "Key not found error in linux-binary-manywheel / manywheel-py3_8-cuda11_8-test / test is unrelated"
